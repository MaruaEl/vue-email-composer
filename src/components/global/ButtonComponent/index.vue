<template>
  <v-btn
    :disabled="isDisabled"
    :type="buttonType"
    :color="setColor"
    :minWidth="minWidth"
    :height="setHeight"
    :to="setRouting"
    class="lighten-2 white--text"
    rounded
  ><slot />{{ buttonText }}</v-btn>
</template>
<script>
export default {
  name: 'ButtonComponent',
  props: {
    buttonText: {
      type: String,
      default: 'Submit',
      required: true,
    },
    buttonType: {
      type: String,
      default: 'button',
      required: false,
    },
    minWidth: {
      type: String,
      default: 'auto',
      required: false,
    },
    setHeight: {
      type: String,
      default: 'auto',
      required: false,
    },
    isDisabled: {
      type: Boolean,
      default: false,
      required: false,
    },
    setColor: {
      type: String,
      default: 'primary',
      required: false,
    },
    setRouting: {
      type: [Boolean, String],
      default: '',
      required: false,
    },
  },
};
</script>
