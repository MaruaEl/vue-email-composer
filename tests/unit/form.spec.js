// Libraries
import Vue from 'vue';
import Vuetify from 'vuetify';

// Utilities
import { mount, createLocalVue } from '@vue/test-utils';

// Components
import ButtonComponent from '@/components/global/ButtonComponent';

const localVue = createLocalVue();

describe('ButtonComponent.vue', () => {
  let vuetify;

  beforeEach(() => {
    vuetify = new Vuetify();
  });

  it('It should have a button text', () => {
    const wrapper = mount(ButtonComponent, {
      localVue,
      vuetify,
      propsData: {
        buttonText: 'Send',
      },
    });

    // With jest we can create snapshot files of the HTML output
    expect(wrapper.html()).toMatchSnapshot();

    // We could also verify this differently
    // by checking the text content
    const title = wrapper.find('button');

    expect(title.text()).toBe('send');
  });

  it('should emit an event when the action v-btn is clicked', () => {
    const wrapper = mount(ButtonComponent, {
      localVue,
      vuetify,
    });

    const event = jest.fn();
    const button = wrapper.find('.v-btn');

    // Here we bind a listener to the wrapper
    // instance to catch our custom event
    // https://vuejs.org/v2/api/#Instance-Methods-Events
    wrapper.vm.$on('action-btn:clicked', event);

    expect(event).toHaveBeenCalledTimes(0);

    // Simulate a click on the button
    button.trigger('click');

    // Ensure that our mock event was called
    expect(event).toHaveBeenCalledTimes(1);
  });
});
