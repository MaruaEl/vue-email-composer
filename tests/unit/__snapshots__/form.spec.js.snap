// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonComponent.vue It should have a button text 1`] = `<v-btn type="button" color="primary" minwidth="auto" height="auto" to="" rounded="" class="lighten-2 white--text">Send</v-btn>`;

exports[`buttonComponent.vue It should have a button text 1`] = `<v-btn type="button" color="primary" minwidth="auto" height="auto" to="" rounded="" class="lighten-2 white--text">Send</v-btn>`;
